lib/omniauth/strategies/realme.rb

```diff
@@ -111,11 +148,45 @@ def saml_settings # rubocop:disable Metrics/AbcSize
 
       private
 
+      ##
+      # Realme documents the various error conditions it can return:
+      #
+      # https://developers.realme.govt.nz/how-realme-works/realme-saml-exception-handling/
+      #
+      def create_exception_for(status_code:, message:) # rubocop:disable Metrics/MethodLength, Metrics/CyclomaticComplexity
+        case status_code
+        when /status:Timeout\z/
+          RealmeTimeoutError.new(message)
+        when /status:InternalError\z/
+          RealmeInternalError.new(message)
+        when /status:AuthnFailed\z/
+          RealmeAuthnFailedError.new(message)
+        when /status:NoAvailableIDP\z/
+          RealmeNoAvailableIDPError.new(message)
+        when /status:NoPassive\z/
+          RealmeNoPassiveError.new(message)
+        when /status:RequestDenied\z/
+          RealmeRequestDeniedError.new(message)
+        when /status:RequestUnsupported\z/
+          RealmeRequestUnsupportedError.new(message)
+        when /status:UnknownPrincipal\z/
+          RealmeUnknownPrincipalError.new(message)
+        when /status:UnsupportedBinding\z/
+          RealmeUnsupportedBindingError.new(message)
+        else
+          RealmeUnrecognisedError.new("Realme login service returned an unrecognised error. status_code=#{status_code} message=#{message}")
+        end
+      end
+
       def allowed_clock_drift
         options.fetch('allowed_clock_drift', 0)
       end
 
-      def default_error_messages(error)
+      def legacy_rails_session_behaviour_enabled?
+        options.fetch('legacy_rails_session_behaviour_enabled', true)
```

Comment:
This option defaults to `true` so existing apps using the gem should be unchanged. In some future version bump, we should change the default to `false`